src/Test/L0/HostContextL0.cs

```diff
@@ -53,71 +53,143 @@ public void GetServiceReturnsSingleton()
             }
         }
 
+        private static readonly (string, string)[] _urlSecretCases = new[]
+        {
+            // some URLs with secrets to mask
+            ("https://user:pass@example.com/path", "https://user:***@example.com/path"),
+            ("http://user:pass@example.com/path", "http://user:***@example.com/path"),
+            ("ftp://user:pass@example.com/path", "ftp://user:***@example.com/path"),
+            ("https://user:pass@example.com/weird:thing@path", "https://user:***@example.com/weird:thing@path"),
+            ("https://user:pass@example.com:8080/path", "https://user:***@example.com:8080/path"),
+            ("https://user:pass@example.com:8080/path\nhttps://user2:pass2@example.com:8080/path", "https://user:***@example.com:8080/path\nhttps://user2:***@example.com:8080/path"),
+            ("https://user@example.com:8080/path\nhttps://user2:pass2@example.com:8080/path", "https://user@example.com:8080/path\nhttps://user2:***@example.com:8080/path"),
+            ("https://user:pass@example.com:8080/path\nhttps://user2@example.com:8080/path", "https://user:***@example.com:8080/path\nhttps://user2@example.com:8080/path"),
+            // some URLs without secrets to mask
+            ("https://example.com/path", null),
+            ("http://example.com/path", null),
+            ("ftp://example.com/path", null),
+            ("ssh://example.com/path", null),
+            ("https://example.com/@path", null),
+            ("https://example.com/weird:thing@path", null),
+            ("https://example.com:8080/path", null),
+        };
```

Comment:
This is a bit complex, sorry, but I wanted to cover combinations of feature flags without repeating the test input.